packages/react-components/src/components/AppFrame/AppFrame.tsx

```diff
@@ -27,6 +27,43 @@ const Frame = (props: IAppFrameProps) => {
   } = props;
   const mergedClassNames = cx(styles[baseClass], className);
   const { isSideNavigationVisible } = useAppFrame();
+  const [isSideNavMounted, setIsSideNavMounted] = React.useState(
+    isSideNavigationVisible
+  );
+  const [isSideNavOpen, setIsSideNavOpen] = React.useState(
+    isSideNavigationVisible
+  );
```

Comment:
I'd suggest to move all animation-related code to the separate hook, so we will keep the code clean and easy ;)